controllers/actions.github.com/ephemeralrunner_controller.go

```diff
@@ -201,7 +202,32 @@ func (r *EphemeralRunnerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		default:
 			// Pod was not found. Create if the pod has never been created
 			log.Info("Creating new EphemeralRunner pod.")
-			return r.createPod(ctx, ephemeralRunner, secret, log)
+			result, err := r.createPod(ctx, ephemeralRunner, secret, log)
+			switch {
+			case err == nil:
+				return result, nil
+			case kerrors.IsInvalid(err) || kerrors.IsForbidden(err):
+				errMessage := fmt.Sprintf("Failed to create the pod: %v", err)
+				if err := r.markAsFailed(ctx, ephemeralRunner, errMessage, ReasonInvalidPodFailure, log); err != nil {
+					log.Error(err, "Failed to set ephemeral runner to phase Failed")
+					return ctrl.Result{}, err
+				}
+				return result, err
```

Comment:
```suggestion
				return ctrl.Result{}, nil
```
Here, if we return an error (which we know is not nil), it will cause another reconciliation. Since we successfully marked an ephemeral runner as failed, we should just return an empty result and a nil so we don't reconcile again

Reply:
👍 
949bb94